src/main/java/mate/academy/dao/impl/MovieDaoImpl.java

```diff
@@ -0,0 +1,48 @@
+package mate.academy.dao.impl;
+
+import java.util.Optional;
+import mate.academy.dao.MovieDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Movie;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+
+@Dao
+public class MovieDaoImpl implements MovieDao {
+    @Override
+    public Movie add(Movie movie) {
+        SessionFactory sessionFactory = HibernateUtil.getSessionFactory();
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.save(movie);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Cannot add movie to DB. Movie: " + movie);
```

Comment:
optional, use another constructor in exception class
```suggestion
            throw new DataProcessingException("Cannot add movie to DB. Movie: " + movie, e);
```